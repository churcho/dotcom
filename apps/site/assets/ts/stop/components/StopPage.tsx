import React, { ReactElement } from "react";
import StopMapContainer from "./StopMapContainer";
import { StopPageData, StopMapData } from "./__stop";
import BreadcrumbContainer from "./BreadcrumbContainer";
import Header from "./Header";
import Sidebar from "./Sidebar";
import AddressBlock from "./AddressBlock";

interface Props {
  stopPageData: StopPageData;
  mapData: StopMapData;
  mapId: string;
}

export default ({
  // eslint-disable-next-line typescript/camelcase
  stopPageData: { stop, routes, tabs, zone_number },
  mapData,
  mapId
}: Props): ReactElement<HTMLElement> => (
  <>
    <BreadcrumbContainer stop={stop} />

    <Header
      stop={stop}
      routes={routes}
      // eslint-disable-next-line typescript/camelcase
      zoneNumber={zone_number}
      tabs={tabs}
    />

    <div className="m-stop-page__details">
      <div className="m-stop-page__details-container">Station Details</div>
    </div>
    <div className="m-stop-page__hero">
      <StopMapContainer initialData={mapData} mapId={mapId} stop={stop} />
      <div className="m-stop-page__hero-photo" />
    </div>
    <div className="container">
      <div className="page-section">
        <div className="row">
          <div className="col-lg-7 col-lg-offset-1">
            Main Column
            <AddressBlock routes={routes} />
          </div>
          <div className="col-lg-4">
            <Sidebar stop={stop} routes={routes} />
          </div>
        </div>
      </div>
    </div>
  </>
);
