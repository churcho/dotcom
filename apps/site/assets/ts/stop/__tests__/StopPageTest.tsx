import React from "react";
import renderer from "react-test-renderer";
import StopPage from "../components/StopPage";
import stopData from "./stopData.json";
import StopPageData from "../components/__stop";

it("it renders", () => {
  const data = JSON.parse(JSON.stringify(stopData)) as StopPageData;

  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';
  const tree = renderer.create(<StopPage stopPageData={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
