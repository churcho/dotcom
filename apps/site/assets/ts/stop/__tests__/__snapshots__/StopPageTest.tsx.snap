// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
Array [
  <div
    className="breadcrumb-container"
  >
    <div
      className="container"
    >
      <span
        className="focusable-sm-down"
      >
        <a
          href="/"
        >
          Home
        </a>
        <i
          aria-hidden="true"
          className="fa fa-angle-right"
        />
      </span>
      <span>
        <a
          href="/stops/subway"
        >
          Stations
        </a>
        <i
          aria-hidden="true"
          className="fa fa-angle-right"
        />
      </span>
      South Station
    </div>
  </div>,
  <div
    className="station__header"
  >
    <div
      className="station__header-container"
    >
      <h1
        className="station__name station__name--upcase"
      >
        South Station
      </h1>
      <div
        className="station__header-features"
      >
        <a
          className="station__header-feature"
          href="/stops/place-sstat?tab=departures#subway-schedule"
        >
          <span
            className="station__header-icon"
          >
            <span
              className="c-svg__icon-red-line-default"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </span>
        </a>
        <a
          className="station__header-feature"
          href="/stops/place-sstat?tab=departures#bus-schedule"
        >
          <span
            className="station__header-icon"
          >
            <span
              className="c-svg__icon-bus-default"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </span>
        </a>
        <a
          className="station__header-feature"
          href="/stops/place-sstat?tab=departures#commuter-rail-schedule"
        >
          <span
            className="station__header-icon"
          >
            <span
              className="c-svg__icon-commuter-rail-default"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </span>
        </a>
        <a
          className="station__header-feature"
          href="/stops/place-sstat?tab=info#commuter-fares"
        >
          <span
            className="station__header-icon c-icon__cr-zone"
          >
            Zone 1A
          </span>
        </a>
        <a
          className="station__header-feature"
          href="#accessibility"
        >
          <span
            className="station__header-icon"
          >
            <span
              className="c-svg__icon-accessible-default"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </span>
        </a>
        <a
          className="station__header-feature"
          href="#parking"
        >
          <span
            className="station__header-icon"
          >
            <span
              className="c-svg__icon-parking-default"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
          </span>
        </a>
      </div>
      <div
        className="header-tabs"
      >
        <a
          className="
      header-tab
      header-tab--dark
      header-tab--selected
      
      details
    "
          href="/stops-v2/place-sstat"
        >
          Station Details
        </a>
        <a
          className="
      header-tab
      header-tab--dark
      
      
      alerts
    "
          href="/stops/place-sstat?tab=alerts"
        >
          Alerts
          <div
            aria-label="1 alert"
            className="m-alert-badge"
          >
            1
          </div>
        </a>
      </div>
    </div>
  </div>,
  <div
    className="station__details"
  >
    <div
      className="station__details-container"
    >
      Station Details
    </div>
  </div>,
  <div
    className="station__hero"
  >
    <div
      className="station__hero-map"
    >
      <div
        className="station__hero-map-container"
        id="test"
      />
      <noscript>
        <style>
          #test { display: none; }
        </style>
        <div
          className="station__hero-map-container"
        >
          <a
            href="https://maps.google.com/?q=700 Atlantic Ave, Boston, MA 02110"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Map of South Station and surrounding area"
              className="map-static map-static-img"
              sizes="(min-width: 1344px) 1099w,
        (min-width: 1088px) 734w,
        (min-width: 800px) 694w,
        calc(100vw - 1.5rem)"
              src=""
              srcSet=""
            />
          </a>
        </div>
      </noscript>
      <div
        className="current-stop"
      />
    </div>
    <div
      className="station__hero-photo"
    />
  </div>,
  <div
    className="container"
  >
    <div
      className="page-section"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-8 col-lg-offset-1"
        >
          Main Column
        </div>
        <div
          className="col-lg-3"
        >
          <p>
            Sidebar
          </p>
          <div
            aria-controls="panel-accessibility"
            aria-expanded={false}
            className="c-expandable-block__header"
            id="header-accessibility"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="c-expandable-block__header-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Accessibility
            <span
              className="c-expandable-block__header-caret"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#xF106;",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
